### PR TITLE
Don't show 'link to globus' when admin browsing a user's account page fixes #1179

### LINKED ISSUE #1179: Don't show 'link to globus' when admin browsing a user's account page
An admin can browse a user's account page, and if the user hasn't linked a globus account, the button 'Link To A Globus Identity' will show up. That button should be hidden. If the admin clicks on it it will make the link with the admin's account anyway.

Change in both CBRAIN and NeuroHub.

### PR DESCRIPTION
**Pull request by MontrealSergiy**
see the attached issue for full description.
Note that the issue presents itself only on CBRAIN part, just in the case the changes done on NeuroHub side too

### CODE REVIEW

File: BrainPortal/app/views/users/show.html.erb [hunk 1]
```diff
@@ -172,13 +172,16 @@
...
                :method => :post,
                :data   => { :confirm => "Are you sure you want to unlink your account with this Globus identity?" }
-             )
-         %>
-       <% else %>
-         (No Globus identity linked to your account)<br>
-         <%= link_to 'Link a Globus identity', @globus_uri, :class => 'button globus_button' %>
-       <% end %>
-    <% end %>
+              )
```
Comment (prioux) [inline BrainPortal/app/views/users/show.html.erb:175]: I don't understand the view logic here.

If the viewing the page as an admin user, we still get the unlink button, which if pressed will not work properly.

Why not just take the view code as it was originally and JUST put a `if current_user.id == @user.id` around the buttons, and nothing else?

Comment (MontrealSergiy) [reply BrainPortal/app/views/users/show.html.erb:175]: The unlink button was not mentioned in the issue.

Comment (prioux) [reply BrainPortal/app/views/users/show.html.erb:175]: I know it wasn't and it's good your thought about it, but the fix is wrong. Just hide it.

Comment (MontrealSergiy) [reply BrainPortal/app/views/users/show.html.erb:175]: sure, putting condition around each button. Then the Globus section cell itself will remain, albeit potentially empty.

Comment (prioux) [reply BrainPortal/app/views/users/show.html.erb:175]: It won't be totally empty, see above, it would say `Provider name:` and `Provider user:`

Comment (prioux) [reply BrainPortal/app/views/users/show.html.erb:175]: Just look at the original code... if you put the if around each button, it works for both normal users and admin users.

**Review by prioux — CHANGES_REQUESTED** [review]

File: BrainPortal/app/views/nh_users/_form.html.erb [hunk 2]
```diff
@@ -132,11 +134,13 @@
       <% end %>
       <% if glob_user.present? %>
         <p class="btn-solid globus bg-warning-bg">
-          <%= link_to nh_unlink_globus_path,
-              :class => "",
-              :method  => :post,
-              :data    => { :confirm => "Are you sure you want to unlink your account from this Globus identity?" } do %>
-              Remove this Globus identity from your NeuroHub account
+          <% if @user.id == current_user.id  # show button only on user own page %>
```
[the remark this answers is not in the document]
Comment (MontrealSergiy) [reply BrainPortal/app/views/nh_users/_form.html.erb:137]: P is moved under if

[the remark this answers is not in the document]
Comment (prioux) [reply BrainPortal/app/views/nh_users/_form.html.erb:137]: Where? That's not what I see in the diff. Look at it.

File: BrainPortal/app/views/nh_users/_form.html.erb [hunk 3]
```diff
@@ -115,8 +115,10 @@
       <% glob_prov = @user.meta[:globus_provider_name] %>
       <% if glob_user.blank? %>
         <p class="btn-solid globus">
-          <%= link_to @globus_uri, :class => "" do %>
-            Connect your NeuroHub account to a Globus identity provider
+          <% if @user.id == current_user.id  # show button only on user own page, hide on other users pages %>
```
Comment (prioux) [inline BrainPortal/app/views/nh_users/_form.html.erb:118]: Same problem here

**Review by prioux — CHANGES_REQUESTED** [review]
I just extracted your PR's code to test it out and discovered that an admin user cannot even look at another user's account page in neurohub. This means that there are no changes needed in the file `nh_users/_form.html.erb`. If you didn't mention this, does that mean you made these modifications without even trying them out yourself?

**Comment by MontrealSergiy** [comment]
It might. Yet I mentioned that in the PR (scroll up to read)

> I just extracted your PR's code to test it out and discovered that an admin user cannot even look at another user's account page in neurohub. This means that there are no changes needed in the file `nh_users/_form.html.erb`. If you didn't mention this, does that mean you made these modifications without even trying them out yourself?

File: BrainPortal/app/views/nh_users/_form.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (MontrealSergiy) [reply BrainPortal/app/views/nh_users/_form.html.erb]: yes, sorry, failed to push. Pushed now. While the bug does not exist on NeuroHub side, one might add user view capability there one day. That person will likely pay no attention to the need to hide button.

**Comment by prioux** [comment]
This looks good now, I'll make a quick check later this afternoon and merge it.

**Comment by prioux** [comment]
Ok, even though on NeuroHub an admin can't look at a user's account page, I'm ok with the changes.

**Pull request merged**